### DISCUSSION TITLE
Google TV/Chromecast VLC alternative

### DISCUSSION THREAD

**Discussion by flipper2222222** [GerryDazoo/Slinger#472] [General]
hi all,

has anyone tried other apps on a Google TV/chromecast other than VLC, interested to hear of your experiences

cheers

**Comment by sp222222** [comment]
i was trying to get mikaplayer working on LG tv but I cannot figure out how to upload the m3u file to it using a webpage.

**Reply by robosmith** [reply]
I've been using MxPlayer which doesn't have the audio/video sync issues of VLC. However it does seem to have more problems dropping streams when the Internet is busy and Slinger can have trouble reconnecting.

**Comment by Mover480** [comment]
I don't think MX Player will work on Google Chromecast TV.  At least I haven't been able to get it to work. Haven't been able to get Kodi to work either.

I've also found that disabling hardware acceleration on VLC fixes the audio/video sync issues.

**Reply by flipper2222222** [reply]
> I don't think MX Player will work on Google Chromecast TV. At least I haven't been able to get it to work. Haven't been able to get Kodi to work either.
> 
> I've also found that disabling hardware acceleration on VLC fixes the audio/video sync issues.

thanks Mover, this setting seems to have mostly eliminated the occasional video buffering I was experiencing

**Reply by flipper2222222** [reply]
I managed to get mx player working on google TV 
in summary
enabled developer mode
downloaded file commander and allowed access to local file system
created m3U play lists in VLC on my PC
copied files via file commander to the Google TV storage, 
opened the M3U file in MXplayer on Google TV , as of now only tested on local network.

**Reply by Mover480** [reply]
Hey flipper, that works!  Thanks for the info.

**Reply by Mover480** [reply]
I verified that this technique also works using the internet (out of LAN viewing).

**Reply by flipper2222222** [reply]
> I verified that this technique also works using the internet (out of LAN viewing).

great, no buffering or any other issues remotely? how many seconds delay when using the remote ?

**Reply by Mover480** [reply]
Seems to work just fine without any buffering.  Remote delay is about 5 to 6 sec which is about the same as on the LAN.

**Reply by flipper2222222** [reply]
thanks for the update, that delay is about the best I get too

**Reply by GerryDazoo** [reply]
Most of that delay is caused by buffering in your media client. Sub-second delays are possible with a client configured for minimal buffering.

**Reply by flipper2222222** [reply]
are there any downsides to reducing the buffering? 
has anyone adjusted VLC or MX player buffering and got the remote delay down to the numbers that Gerry has quoted above?

**Reply by GerryDazoo** [reply]
Many moons ago on some version of VLC I came across a checkbox for "minimum delay" on VLC and it worked OK. But I have not been able to find it ever since. I use ffplay.exe to get the low-latency.
i.e. C:\ffmpeg\ffplay.exe -infbuf -probesize 32 -analyzeduration 0 -vf scale=1280x720 -flags nobuffer -flags low_delay http://192.168.117.113:65432/slingbox

Is there a downside? Not that I've seen. You'd have to be on a very crappy network with huge latencies before you'd get problems. If you do just bump up the probesize until it works.

Slinger Player is coming soon with the low-latency config as the default.

**Comment by sp222222** [comment]
how do you create the m3u file?
what are those steps?
I’m trying to get a file like that onto a LG smart tv

**Reply by blasco403** [reply]
Create text file with your desired name. Make sure it has the file extension .m3u

xxx.m3u

Content is a single line with the URL you use to connect to your slingbox via slinger.

http://domain:port/baseURL/name

Move/copy file to a location on the device you are using. Open xxx.m3u file with your desired stream client.

**Reply by flipper2222222** [reply]
I followed this 
https://www.videoconverterfactory.com/tips/create-m3u-playlist.html

let us know if you get it on to the LG , I have one too
